src/main/java/org/junitpioneer/jupiter/DefaultLocaleExtension.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.util.Locale;
+import java.util.Optional;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+class DefaultLocaleExtension implements BeforeAllCallback, BeforeEachCallback, AfterAllCallback, AfterEachCallback {
+
+	private Locale defaultLocale;
```

Comment:
I have not tested this in an IDE, but storing this state in a field within the extension looks brittle.

I think this needs to be stored in the `ExtensionContext.Store` for the current execution level (i.e., class or method).

Reply:
I talked to this with @sormuras and we agreed on the simpler approach. But I can change it to `ExtensionContext.Store`. WDYT @sormuras?

Reply:
I was taught (and am teaching people) to always use the extension store as there is no guarantee for an extension instance's lifecycle. In case I'm missing something, I'd also prefer the store.

Reply:
_"Use the Store, Luke!"_